src/updateFileFromServer.ts

```diff
@@ -73,12 +73,36 @@ export async function toggleServerTaskStatus(e: Editor, settings: TodoistSetting
 	}
 }
 
-
-async function getServerData(todoistQuery: string, authToken: string): Promise<string> {
+async function getServerData(todoistQuery: string, authToken: string, showSubtasks: boolean): Promise<string> {
 	const api = new TodoistApi(authToken)
+
+	const tasks = await callTasksApi(api, todoistQuery);
+
+	let subtasks: Task[];
+	if (showSubtasks) {
+		subtasks = await callTasksApi(api, 'subtask');
```

Comment:
i'm a bit surprised to see this line of code. This ends up pulling all the subtasks for every project, even if the user has a project that this plugin would never want to touch.

This could also be more likely to have performance issues - the set of all subtasks may be much larger than the set of subtasks that the user cares to see.

Is there a way to change this to only pull the relevant subtasks? It looks like when we do the original query, it does include subtask results by default. So perhaps we can just use the value of the `parent_id` field to figure out what is a subtask and what isn't

Reply:
If there is a way to limit down to only the relevant ones, I haven't been able to find it.  The performance concern was top of the list for me as well.  I was searching various threads and banging around in Postman for quite a bit trying to get a combination to work, but didn't find any joy.

Running the today|overdue filter (same tasks as I used in this PR for testing) returns the below Task detail.  No indication of subtasks.  I get the same object if I hit the API for the specific task ID, too.

``` json
[
    {
        "id": "6465138149",
        "assigner_id": null,
        "assignee_id": null,
        "project_id": "2266227575",
        "section_id": null,
        "parent_id": null,
        "order": 76,
        "content": "Subtasks Testing - Top Level",
        "description": "",
        "is_completed": false,
        "labels": [],
        "priority": 4,
        "comment_count": 0,
        "creator_id": "34056777",
        "created_at": "2022-12-25T18:57:09.31511Z",
        "due": {
            "is_recurring": false,
            "string": "Dec 31",
            "date": "2022-12-31"
        },
        "url": "https://todoist.com/showTask?id=6465138149"
    }
]
```
My starting thought was to fire API request(s) to get the subtasks associated for parent_IDs that came back from the today|overdue filter.  But I couldn't seem to convince the API to see it my way.  I also did try various combinations of the filters, including: 

* parent_id=6465138149
  * 400 Bad Request - There's a problem with your filter: "Unrecognized symbol parent_id=6465138149"
* parent=6465138149
  * 400 Bad Request - There's a problem with your filter: "Unrecognized symbol parent=6465138149"
* today|overdue|subtask
  * just returned a combined list of all the things, didn't filter.
* today|overdue & subtask
  *  only returned that same top level task shown in the JSON above.
* (today|overdue) & subtask
  * Empty array of tasks.  I think it's just getting cross with me at this point.

I should have detailed out what I had tried more in the initial PR, sorry about.  So far, I keep finding myself limited by the constraints of what I was able to get working with the API.  Let me know your thoughts on next steps here.

Reply:
Hmm, so this is different from what I observed.

I have the following
<img width="320" alt="Screen Shot 2023-01-01 at 10 52 39 AM" src="https://user-images.githubusercontent.com/16907831/210176871-6baa78d5-f4b4-4d6d-9835-d37b0caca740.png">

and I ran 

```
 /usr/bin/curl -X GET \
  "https://api.todoist.com/rest/v2/tasks?filter=%23Inbox" \
  -H "Authorization: Bearer <REDACTED>"
[
	{
		"id": "6480113598",
		"assigner_id": null,
		"assignee_id": null,
		"project_id": "2215750670",
		"section_id": null,
		"parent_id": null,
		"order": 1,
		"content": "test root",
		"description": "",
		"is_completed": false,
		"labels": [],
		"priority": 1,
		"comment_count": 0,
		"created_at": "2023-01-01T15:52:09.315206Z",
		"due": null,
		"url": "https://todoist.com/showTask?id=6480113598"
	},
	{
		"id": "6480113786",
		"assigner_id": null,
		"assignee_id": null,
		"project_id": "2215750670",
		"section_id": null,
		"parent_id": "6480113598",
		"order": 1,
		"content": "test subtask",
		"description": "",
		"is_completed": false,
		"labels": [],
		"priority": 1,
		"comment_count": 0,
		"created_at": "2023-01-01T15:52:16.397444Z",
		"due": null,
		"url": "https://todoist.com/showTask?id=6480113786"
	}
]
```

so you could observe the child-to-parent relationship here.

Perhaps you had a mismatched due date on your child task?

Reply:
Okay, your response certainly helped me figure out what was going on.  Thanks!

Up to this point, I only use the default query/filter of the plugin (today/overdue).  I did indeed set a date on the parent task I have been testing with to pull it into that query.  However, none of the subtasks I created had due dates.  Because of this, I wasn't seeing any subtasks when I was testing.  Adding the missing due date to the subtasks did indeed cause it to return with the filter.  In other words, due date based filters exclude tasks with null due dates.

With your example, using the Inbox filter pulled everything, including tasks with null due dates.  My first (incorrect) take on this was that it needs the call to the subtasks filter even more at that point in order to get a true representation of all associated subtasks, since the different filters may or may not bring back all the subtasks.  But quickly I realized, nope, they were filtered for a reason.

Many of my Todoist entries with subtasks are recurring as reminders of things to do.  For example: do thing X is the parent task, and there are 3 subtasks of additional specific things to help me remember.  This looks right in the Todoist app (only one task shown in the Today view along with a count of subtasks).  However, the REST API doesn't seem to work in quite the same way.

My next thought was that if I want all the recurring subtasks to show up properly using the today/overdue filter, the subtasks need to have the same recurring option.  Tried that, and it does pull all the subtasks via the API with a date filter, but now they're also top level items (rightfully so since now those tasks have a today due date).  This also makes for some serious ugly duplication of the task lasting in the existing code, as it assumed the top level tasks came from the first query.  Argh!

It's all solvable, but I'll need to fiddle around with it.  More to come!

Reply:
Makes sense! it's definitely an odd situation. My leaning would be to get as close to the official web client as much as possible. If there's multiple options, feel free to write them there to discuss before implementing (if they're long/complex implementations)

Reply:
I actually pushed a commit last night that I think got everything sorted out on this one!  Forgot to update this comment thread.